fix: same $HOME between install and uninstall

netanel-haber opened the pull request:
fixes issue https://github.com/datreeio/datree/issues/783.

netanel-haber committed:
fix: same $HOME between install and uninstall

fixes issue https://github.com/datreeio/datree/issues/783.

hadar-co commented on install.sh:
```diff
@@ -1,18 +1,20 @@
     echo "This script must be executed with root privileges." && exit 1
     fi
     rm -f /usr/local/bin/datree
-    rm -rf $HOME/.datree
```
Why move these 2 lines out of the `cat` section? You can leave them in as:
```
rm -rf "$DATREE_HOME"
echo "Datree was successfully uninstalled."
```

netanel-haber replied:
Thats exactly what this PR fixes - the tricky thing with the `cat >> X.sh << 'END' ... END` trick is that it outputs **the literal text** into the file, without interpolation. My fix ensures that the initial install.sh $HOME is captured, and input into the uninstall.sh file.

[This video](https://user-images.githubusercontent.com/58652339/189615325-400cd1f9-c350-4c8c-98fb-0a8c8bb0d63c.mov) shows that $HOME isn't evaluated when catting the string, rather `$HOME` is input **literally** into the file, which is the exact problem in the first place.

So in your suggestion, the string `$DATREE_HOME` will be input into the file - and this variable obviously won't be available when the script is run.

hadar-co replied:
Yup, I see what you mean. Thanks for the detailed response👍🏼

hadar-co replied:
@netanel-haber 
After some reading&testing, it seems that the fix is much simpler. The `cat >> X.sh << END ... END` (AKA [here-document](https://tldp.org/LDP/abs/html/here-docs.html)) does interpolate variables by default. To disable the interpolation, you add quotes around the `END`, which is what we originally had in the script, so we should simply remove them.
If you want to prevent a single instance from being interpolated - like in our case the `$(id -u)`, you can simply escape it using `\`. Put all of this together and the fix should look like this:
```
UNINSTALL_SCRIPT="$HOME/.datree/uninstall.sh"
    touch $UNINSTALL_SCRIPT && chmod +x $UNINSTALL_SCRIPT

    cat > $UNINSTALL_SCRIPT << END
    if [ "\$(id -u)" -ne 0 ] ; then
    echo "This script must be executed with root privileges." && exit 1
    fi
    rm -f /usr/local/bin/datree
    rm -rf $HOME/.datree
    echo "Datree was successfully uninstalled."
END
``` 

Bash is tricky 😈

@myishay FYI

myishay replied:
@hadar-co Interesting!

@netanel-haber WDYT?

netanel-haber replied:
Much nicer. Learned something too.

myishay replied:
@netanel-haber So would you like to make the desired changes? I think we can merge after that 🚀

netanel-haber replied:
Sure

hadar-co commented on install.sh:
```diff
@@ -71,7 +73,7 @@ echo -e "\033[32m[V] Finished Installation\033[0m"
 echo
 
-echo -e "\033[35m Usage: $ datree test ~/.datree/k8s-demo.yaml \033[0m"
+echo -e "\033[35m Usage: $ datree test $DATREE_HOME/k8s-demo.yaml \033[0m"
```
This is output for the user, should stay as `~` so it's shorter

hadar-co commented:
hey @netanel-haber! thanks for the PR, left some comments:)

netanel-haber commented:
> hey @netanel-haber! thanks for the PR, left some comments:)

I don't see any comments?

hadar-co commented:
> > hey @netanel-haber! thanks for the PR, left some comments:)
> 
> I don't see any comments?

My bad, try now:)

netanel-haber committed:
PR comments: User facing log should still use ~

PR comments: User facing log should still use ~ instead of expanded variable.

netanel-haber committed:
unwrap apostrophe from here-document head label

this enables parameterization. Also: escaping `\$id -u` so it is input literally.

netanel-haber committed:
revert noise
